api/client/commands.go

```diff
@@ -769,26 +779,19 @@ func (cli *DockerCli) CmdStart(args ...string) error {
 		}
 	}
 
-	var encounteredError error
 	for _, name := range cmd.Args() {
 		_, _, err := readBody(cli.call("POST", "/containers/"+name+"/start", nil, false))
 		if err != nil {
 			if !*attach && !*openStdin {
 				fmt.Fprintf(cli.err, "%s\n", err)
 			}
-			encounteredError = fmt.Errorf("Error: failed to start one or more containers")
+			return fmt.Errorf("Error: failed to start one or more containers")
```

Comment:
sorry, I made a mistake here, return should't be here, since docker start can start multiple containers, this will break the start if a container start failed. PR #11225 is going to fix this. Sorry for the mistake. ping @tiborvass  @tiborvass